dvc/testing/test_api.py

```diff
@@ -18,3 +21,103 @@ def test_open(self, tmp_dir, dvc, remote):
 
         with api.open("foo") as fd:
             assert fd.read() == "foo-text"
+
+    @pytest.mark.parametrize(
+        "clear_cache", [True, False], ids=["cache", "no_cache"]
+    )
+    @pytest.mark.parametrize(
+        "fs_kwargs",
+        [
+            {},
+            {"url": "{path}"},
+            {"url": "{path}", "rev": "master"},
+            {"url": "file://{posixpath}"},
+            {"url": "file://{posixpath}", "rev": "master"},
+        ],
+        ids=["current", "local", "local_rev", "git", "git_rev"],
+    )
+    def test_filesystem(
+        self,
+        M,
```

Comment:
I remember discussing this before, but `M` is completely unreadable unless you already know what this is. The name is not descriptive at all and searching for it to learn more is hell 🙂

Reply:
No action needed, just ranting.

Reply:
yeah, there's `Matcher` alias too that we can use instead.
But after you know what `M` is, the `Matcher.dict` feels too lengthy. :)

Reply:
Just a heads up that this breaks CI in all of our FS plugins since none of them import pytest-test-utils right now. We should probably have a `dvc[testing]` extra that pulls in everything that `dvc/testing` needs.

Reply:
and yeah it took me a while to figure out where the `M` fixture was even supposed to come from.

Reply:
> and yeah it took me a while to figure out where the M fixture was even supposed to come from.

That's the "magic" of pytest fixture dependency injection.

Reply:
:sob: `M` is hell

Reply:
> 😭 M is hell

It does match the Majin Mark from Dragon Ball. 

![imagen](https://user-images.githubusercontent.com/12677733/190671941-cd13ea07-082a-4346-9f40-2ed321f1f0b7.png)